cli/smartcontract/smart_contract.go

```diff
@@ -646,12 +652,16 @@ func contractDeploy(ctx *cli.Context) error {
 		return cli.NewExitError(err, 1)
 	}
 
-	txScript, err := request.CreateDeploymentScript(avm, &conf.Contract)
+	m := conf.ToManifest(avm)
+	w := io.NewBufBinWriter()
+	m.EncodeBinary(w.BinWriter)
+	rawManifest := w.Bytes()
+	txScript, err := request.CreateDeploymentScript(avm, rawManifest)
```

Comment:
Why `rawManifest`? I think it'd be more convenient to pass normal manifest. Maybe even as Serializable. And we could avoid some excessive buffering that way.

Reply:
Initally that was to not marshal manifest twice. But `contractDeploy` won't be called frequently, so this is not a big deal I suppose.

Reply:
Fixed.